modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -0,0 +1,676 @@
+
+                // FIXME: THIS ORDER IS IRRELEVANT TO PROTOCOL OR ANY OTHER ORDER!
+                // FIXME: SAME APPLIES TO THE REGULAR GEEXECUTOR!!
+                auto xtract_in = [&](ade::NodeHandle slot_nh, std::vector<RcDesc> &vec) {
+                    const auto orig_data_nh
+                        = m_gim.metadata(slot_nh).get<DataSlot>().original_data_node;
+                    const auto &orig_data_info
+                        = m_gm.metadata(orig_data_nh).get<Data>();
+                    if (orig_data_info.storage == Data::Storage::CONST_VAL) {
+                        const_ins.insert(slot_nh);
+                        // FIXME: Variant move issue
+                        in_constants.push_back(const_cast<const cv::GRunArg&>(m_gm.metadata(orig_data_nh).get<ConstValue>().arg));
+                    } else in_constants.push_back(cv::GRunArg{});
+                    if (orig_data_info.shape == GShape::GARRAY) {
+                        GAPI_Assert(orig_data_info.ctor.index() != orig_data_info.ctor.index_of<cv::util::monostate>());
+                    }
+                    vec.emplace_back(RcDesc{ orig_data_info.rc
+                                           , orig_data_info.shape
+                                           , orig_data_info.ctor});
+                };
+                auto xtract_out = [&](ade::NodeHandle slot_nh, std::vector<RcDesc> &vec, cv::GMetaArgs &metas) {
+                    const auto orig_data_nh
+                        = m_gim.metadata(slot_nh).get<DataSlot>().original_data_node;
+                    const auto &orig_data_info
+                        = m_gm.metadata(orig_data_nh).get<Data>();
+                    if (orig_data_info.shape == GShape::GARRAY) {
+                        GAPI_Assert(orig_data_info.ctor.index() != orig_data_info.ctor.index_of<cv::util::monostate>());
+                    }
+                    vec.emplace_back(RcDesc{ orig_data_info.rc
+                                           , orig_data_info.shape
+                                           , orig_data_info.ctor});
+                    metas.emplace_back(orig_data_info.meta);
+                };
+                // FIXME: JEZ IT WAS SO AWFUL!!!!
+                for (auto in_slot_nh  : nh->inNodes())  xtract_in(in_slot_nh,  input_rcs);
+                for (auto out_slot_nh : nh->outNodes()) xtract_out(out_slot_nh, output_rcs, output_metas);
+
+                m_ops.emplace_back(OpDesc{ std::move(input_rcs)
+                                         , std::move(output_rcs)
+                                         , std::move(output_metas)
+                                         , nh
+                                         , in_constants
+                                         , m_gim.metadata(nh).get<IslandExec>().object});
+
+                // Initialize queues for every operation's input
+                ade::TypedGraph<DataQueue> qgr(*m_island_graph);
```

Comment:
You can create this above, to not create twice (`NodeKind::SINK`)

Reply:
I believe the lifetime of a variable should be as short as possible. Creating a TypedGraph is cheap. Having it declared only here means it is accessed only here.